lib/util.py

```diff
@@ -542,55 +542,62 @@ def format_time(timestamp):
+    else:
+        return _("in {time}").format(time=td)
 
 
 def time_difference(distance_in_time, include_seconds):
-    #distance_in_time = since_date - from_date
+    #distance_in_time = from_date - since_date
     distance_in_seconds = int(round(abs(distance_in_time.days * 86400 + distance_in_time.seconds)))
-    distance_in_minutes = int(round(distance_in_seconds/60))
+    distance_in_minutes = int(round(distance_in_seconds / 60))
 
-    if distance_in_minutes <= 1:
+    if distance_in_seconds < 60:
         if include_seconds:
             for remainder in [5, 10, 20]:
                 if distance_in_seconds < remainder:
-                    return "less than %s seconds" % remainder
+                    return _("less than {seconds} seconds").format(seconds=remainder)
             if distance_in_seconds < 40:
-                return "half a minute"
-            elif distance_in_seconds < 60:
-                return "less than a minute"
+                return _("half a minute")
             else:
-                return "1 minute"
+                return _("less than a minute")
         else:
-            if distance_in_minutes == 0:
-                return "less than a minute"
-            else:
-                return "1 minute"
+            return _("less than a minute")
     elif distance_in_minutes < 45:
-        return "%s minutes" % distance_in_minutes
+        fmt = ngettext("{minutes} minute", "{minutes} minutes", distance_in_minutes)
+        return fmt.format(minutes=distance_in_minutes)
     elif distance_in_minutes < 90:
-        return "about 1 hour"
+        return _("about 1 hour")
     elif distance_in_minutes < 1440:
-        return "about %d hours" % (round(distance_in_minutes / 60.0))
-    elif distance_in_minutes < 2880:
-        return "1 day"
+        distance_in_hours = round(distance_in_minutes / 60.0)
+        return _("{hours} hours").format(hours=distance_in_hours)
+    elif distance_in_minutes < 2160:
+        return _("about 1 day")
     elif distance_in_minutes < 43220:
-        return "%d days" % (round(distance_in_minutes / 1440))
-    elif distance_in_minutes < 86400:
-        return "about 1 month"
+        distance_in_days = round(distance_in_minutes / 1440.0)
+        return _("{days} days").format(days=distance_in_days)
+    elif distance_in_minutes < 64830:
+        return _("about 1 month")
     elif distance_in_minutes < 525600:
-        return "%d months" % (round(distance_in_minutes / 43200))
-    elif distance_in_minutes < 1051200:
-        return "about 1 year"
+        distance_in_months = round(distance_in_minutes / 43200.0)
+        return _("{months} months").format(months=distance_in_months)
+    elif distance_in_minutes < 788400:
+        return _("about 1 year")
     else:
-        return "over %d years" % (round(distance_in_minutes / 525600))
+        distance_in_years = round(distance_in_minutes / 525600.0)
+        return _("{years} years").format(years=distance_in_years)
```

Comment:
`ngettext` here?

Reply:
Good job on removing the 'over'.. but since it's rounded I also suggest doing `about {years} years` here...

Reply:
I had that first, but then I realized it can not ever get there with only 1 year. Though now that I think about it again, there can be more plural forms, so we should have it. I'll update it.

Reply:
I'd like to have the "about" only in the range between 1 to 1.5 years. Maybe even shorter, right now all the about are a factor of 1.5.

Reply:
ok